pkg/sql/alter_table.go

```diff
@@ -353,14 +353,9 @@ func (n *alterTableNode) startExec(params runParams) error {
 			case *tree.CheckConstraintTableDef:
 				var err error
 				params.p.runWithOptions(resolveFlags{contextDatabaseID: n.tableDesc.ParentID}, func() {
-					info, infoErr := n.tableDesc.GetConstraintInfo()
-					if infoErr != nil {
-						err = infoErr
-						return
-					}
 					ckBuilder := schemaexpr.MakeCheckConstraintBuilder(params.ctx, *tn, n.tableDesc, &params.p.semaCtx)
-					for k := range info {
-						ckBuilder.MarkNameInUse(k)
+					for _, c := range n.tableDesc.AllConstraints() {
+						ckBuilder.MarkNameInUse(c.GetName())
```

Comment:
This is one of these instances where we're more careful to not re-use a name. Things can get quite confusing otherwise when rolling back a schema change involving a name collision.